src/server/pfs/server/source.go

```diff
@@ -199,6 +205,32 @@ func computeFileHash(idx *index.Index) []byte {
 	return h.Sum(nil)
 }
 
+type errOnEmpty struct {
+	source Source
+	err    error
+}
+
+// InsertNotFound causes iterate to return a not found error if there are no items to iterate over
+func ErrOnEmpty(s Source, err error) Source {
```

Comment:
`NewErrOnEmpty` if you are sticking with an abstraction rather than a helper function. Also, change the comment or the linter won't let you pass.